### ISSUE TITLE
How to contribute via GitLab

### ISSUE THREAD

**Issue by vlcinsky** [timvisee/ffsend#118]
Current documentation is proposing to contribute via GitLab.

I tried that but failed. Using feature branch my push to GitLab project repo was rejected with "You are not allowed to push code to this project".

Shall I clone the repo to my own account? Will the merge request work then?

I am also confused by the fact, issues and code lives on two places. It is very likely it generates extra syncing effort and is error prone.

Proposal:
- either decide to use only GitHub (simpler for getting contributions)
- or explain in `CONTRIBUTING.md` how shall be the Merge request done on GitLab.

**Comment by timvisee** [comment]
Yes, merge requests essentially work the same on GitLab as pull requests on GitHub. You've to fork the repository to your own account, make changes, then submit a merge request.

I prefer hosting on GitLab, which is why I moved the project over there (more than a year ago). I'm keeping this repository on GitHub live as a mirror, for various reasons. Many are still using this GitHub version as source with automation tools and whatnot so I don't want to close this down. The social aspect of GitHub is also much better, providing better publicity for `ffsend`.

Now, the main issue board is on GitLab as well, as the main repository lives there. Some users mentioned that they preferred to allow posting issues on GitHub as well as some don't have a GitLab account. That's why I decided to keep both issue boards open. The same goes for pull requests by the way.

I manually migrate any significant issues over to GitLab. So everything important should always be on the GitLab side of things. I agree this isn't the most elegant approach, but since the number of posted issues is limited I don't think this is a big problem. Sadly GitHub doesn't allow to link the Issue tab on the top of the repository page to an external page, that would arguably make the flow of issue posting on GitLab from here better.

There is actually a problem with the current set up. Release binaries are on GitHub, not on GitLab, so yes that's weird. I didn't take the time yet to migrate this to GitLab through automation. So that's something that still needs to be done.

To summarize:
- I want to use GitLab primarily
- I like to keep the GitHub mirror alive (compatibility/publicity)
- I'm not sure about closing the issue board on GitHub, currently left it open as well as a middle ground
- Release binaries should still be migrated to GitLab

Maybe linking the GitLab issue board in the template for creating a new issue on GitHub here should improve things. Maybe I should close the GitHub issue board down. I don't know. I'm not sure what is best to do here.
